spec.bs

```diff
@@ -995,6 +995,74 @@ To <dfn>escape a regexp string</dfn> given a string |input|:
+
+  1. Let |result| be the empty string.
+  1. [=list/For each=] |part| of |part list|:
+    1. If |part|'s [=part/type=] is "<a for=part/type>`fixed-text`</a>" then:
+      1. If |part|'s [=part/modifier=] is "<a for=part/modifier>`none`</a>" then:
+        1. Append the result of running [=escape a pattern string=] given |part|'s [=part/value=] to the end of |result|.
+        1. [=Continue=].
+      1. Append "`{`" to the end of |result|.
+      1. Append the result of running [=escape a pattern string=] given |part|'s [=part/value=] to the end of |result|.
+      1. Append "`}`" to the end of |result|.
+      1. Append the result of running [=convert a modifier to a string=] given |part|'s [=part/modifier=] to the end of |result|.
+      1. [=Continue=].
+    1. Let |needs grouping| be true if at least one of the following are true and otherwise false:
+      <ul>
+        <li>|part|'s [=part/suffix=] is not the empty string.</li>
+        <li>|part|'s [=part/prefix=] is not the empty string and is not |options|'s [=options/prefix code point=].
+      </ul>
+    1. [=Assert=]: |part|'s [=part/name=] is not the empty string or null.
+    1. Let |custom name| be true if |part|'s [=part/name=][0] is not an [=ASCII digit=].
+    1. If |needs grouping| is true, then append "`{`" to the end of |result|.
+    1. Append the result of running [=escape a pattern string=] given |part|'s [=part/prefix=] to the end of |result|.
+    1. If |custom name| is true:
+      1. Append "`:`" to the end of |result|.
+      1. Append |part|'s [=part/name=] to the end of |result|.
+    1. If |part|'s [=part/type=] is "<a for=part/type>`regexp`</a>" then:
+      1. Append "`(`" to the end of |result|.
+      1. Append |part|'s [=part/value=] to the end of |result|.
+      1. Append "`)`" to the end of |result|.
+    1. Else if |part|'s [=part/type=] is "<a for=part/type>`segment-wildcard`</a>" and |custom name| is false:
+      1. Append "`(`" to the end of |result|.
+      1. Append the result of running [=generate a segment wildcard regexp=] given |options| to the end of |result|.
+      1. Append "`)`" to the end of |result|.
+    1. Else if |part|'s [=part/type=] is "<a for=part/type>`full-wildcard`</a>":
+      1. If |custom name| is true:
+        1. Append "`(`" to the end of |result|.
+        1. Append [=full wildcard regexp value=] to the end of |result|.
+        1. Append "`)`" to the end of |result|.
+      1. Else append "`*`" to the end of |result|.
+    1. Append the result of running [=escape a pattern string=] given |part|'s [=part/suffix=] to the end of |result|.
+    1. If |needs grouping| is true, then append "`}`" to the end of |result|.
+    1. Append the result of running [=convert a modifier to a string=] give |part|'s [=part/modifier=] to the end of |result|.
+  1. Return |result|.
+</div>
+
+<div algorithm>
+To <dfn>escape a pattern string</dfn> given a string |input|:
```

Comment:
You could unify this algorithm with "escape a regexp string" by having it take a list of escaped characters as an input, and defining the regexp escaped code points list and the pattern string escaped code points list. Similar to percent encoding/decoding and the various "percent-encode sets" in https://url.spec.whatwg.org/#percent-encoded-bytes .

I'd definitely do that if you have a third instance of this sort of algorithm. With only two it's less clear it would be worth it.

Reply:
I have them unified in the c++.  I guess I'd like to wait to unify them when I restructure and clean up the doc.  I think I'll end up creating a "utilities" section where they will be co-located and would make more sense then.  Going to leave as is for now, though.